odpf/shield/v1/shield.proto

```diff
@@ -0,0 +1,546 @@
+syntax = "proto3";
+package odpf.shield.v1;
+
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "validate/validate.proto";
+
+option go_package = "github.com/odpf/proton/shield/v1;shieldv1";
+option java_package = "io.odpf.proton.shield.v1";
+option java_outer_classname = "Shield";
+
+
+// These annotations are used when generating the OpenAPI file.
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+  info: {
+    title: "Shield",
+    version: "0.1.0";
+  };
+  schemes: HTTP;
+};
+
+service ShieldService {
+  // Users
+  rpc GetUsers(GetUsersRequest) returns (GetUsersResponse) {
+    option (google.api.http) = {
+      get: "/v1/users"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "User";
+      summary: "Get All Users";
+    };
+  }
+  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {
+    option (google.api.http) = {
+      post: "/v1/users",
+      body: "body";
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "User";
+      summary: "Create User";
+    };
+  }
+
+  rpc GetUserByID(GetUserByIDRequest) returns (GetUserByIDResponse) {
```

Comment:
Let's call this `GetUser`

Reply:
The reason I am asking is that we should be able to utilize the same method in future for other GetUser requests as well on the basis of field that is set.
And if required we can add aliases for the http endpoints.